recipes/lora_finetune_distributed.py

```diff
@@ -563,6 +680,7 @@ def train(self) -> None:
         running_loss = 0
         num_tokens = 0
 
+        self._profiler.start()
```

Comment:
It's really nice how cleanly this fits into our training loop. This is exactly what I was hoping for